app/jobs/application_job.rb

```diff
@@ -36,6 +36,8 @@ class ApplicationJob < ActiveJob::Base
   # @param [Hash<Symbol => String>] metadata
   # @option metadata [String] :checksum_md5
   # @option metadata [Integer] :size
+  # @option metadata [Integer] :parts_count
+  # @option metadata [String] :suffix
```

Comment:
No, you can always refer from one piece of yard to another rather than copy it, like:
```ruby
@see OtherClass#method
```
And you cannot pass an arbitrary class as an argument to a Job, because it has to have serialization and deserialization built in (since that is what allows asynchronicity), so the proposed class is incompatible with the required use.